MovieApp/Infrastructure/ViewModels/MovieDetailsViewModel.swift

```diff
@@ -0,0 +1,31 @@
+import Foundation
+
+class MovieDetailsViewModel {
+    var movieDetails: MovieService.MovieDetailsResponse? {
+        didSet {
+            onUpdateMovieDetails?()
+        }
+    }
+    
+    var onUpdateMovieDetails: (() -> Void)?
+    private let movieId: Int
+    private let dataSource = MovieService()
+    
+    init(movieId: Int) {
+        self.movieId = movieId
+        fetchMovieDetails()
+    }
+    
+    func fetchMovieDetails() {
+        Task {
+            do {
+                let details = try await dataSource.fetchMovieDetails(for: movieId)
+                DispatchQueue.main.async {
+                    self.movieDetails = details
+                }
+            } catch {
+                print("Error fetching movie details: \(error)")
+            }
+        }
+    }
```

Comment:
Same approach could be used here as mentioned in one of the previous comments about threading. Please check out how @MainActor can be utilised to execute tasks on the main thread.